ui/packages/components/src/CodeBlock/CommandBlock.tsx

```diff
@@ -0,0 +1,180 @@
+'use client';
+
+import { useEffect, useRef, useState } from 'react';
+import { CopyButton } from '@inngest/components/CopyButton';
+import { useCopyToClipboard } from '@inngest/components/hooks/useCopyToClipboard';
+import { cn } from '@inngest/components/utils/classNames';
+import { FONT, LINE_HEIGHT, createColors, createRules } from '@inngest/components/utils/monaco';
+import Editor, { useMonaco } from '@monaco-editor/react';
+import * as Tabs from '@radix-ui/react-tabs';
+import { type editor } from 'monaco-editor';
+
+import { isDark } from '../utils/theme';
+
+export type TabsProps = {
+  title: string;
+  content: string;
+  readOnly?: boolean;
+  language?: string;
+};
+
+type MonacoEditorType = editor.IStandaloneCodeEditor | null;
+
+const CommandBlock = ({ currentTabContent }: { currentTabContent?: TabsProps }) => {
+  const [dark, setDark] = useState(isDark());
+  const editorRef = useRef<MonacoEditorType>(null);
+  const wrapperRef = useRef<HTMLDivElement>(null);
+
+  const monaco = useMonaco();
+  const activeTabContent = currentTabContent || { content: '', readOnly: true, language: 'json' };
+
+  useEffect(() => {
+    // We don't have a DOM ref until we're rendered, so check for dark theme parent classes then
+    if (wrapperRef.current) {
+      setDark(isDark(wrapperRef.current));
+    }
+  });
+
+  useEffect(() => {
+    if (!monaco) {
+      return;
+    }
+
+    monaco.editor.defineTheme('inngest-theme', {
+      base: dark ? 'vs-dark' : 'vs',
+      inherit: true,
+      rules: dark ? createRules(true) : createRules(false),
+      colors: dark ? createColors(true) : createColors(false),
+    });
+  }, [monaco, dark]);
+
+  const handleEditorDidMount = (editor: MonacoEditorType) => {
+    editorRef.current = editor;
+    updateEditorHeight();
+  };
+
+  const updateEditorHeight = () => {
+    const editor = editorRef.current;
+    if (editor) {
+      const contentHeight = Math.min(1000, editor.getContentHeight());
+      wrapperRef.current!.style.height = `${contentHeight}px`;
+      editor.layout();
+    }
+  };
+
+  return (
+    <>
```

Comment:
These extra fragment wrappers don't seem necessary...more content coming soon?

Reply:
There's one there that can go, was a leftover from a failed experiment with some monaco height stuff